### PR TITLE
Feat/application parse data

### PR DESCRIPTION
**Pull request by dannil76**

### CODE REVIEW

File: app/schemas/period_schema.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+from marshmallow import Schema as SimpleSchema
+from marshmallow_jsonapi import fields
```
Comment (TeddyAndersson) [inline app/schemas/period_schema.py:2]: import fields from marshmallow and remove dependecy for marshmallow_jsonapi

File: app/schemas/response_schema.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
+from marshmallow import Schema as SimpleSchema
+from marshmallow_jsonapi import fields
```
Comment (TeddyAndersson) [inline app/schemas/response_schema.py:2]: import fields from marshmallow and remove dependecy for marshmallow_jsonapi

File: app/routes/applications.py [hunk 3]
```diff
@@ -1,28 +1,89 @@
...
 
-parser = reqparse.RequestParser()
-parser.add_argument('application_type', type=str, required=True)
-parser.add_argument('user_hash', type=str, required=True)
-parser.add_argument('data', type=dict, required=True)
+
+cost_types = ['EXPENSES', 'INCOMES']
+
+
+def parse_application_data(data, period, parent_key=None, initial_data=dict):
```
Comment (TeddyAndersson) [inline app/routes/applications.py:12]: Move to libs folder

File: app/schemas/application_schema.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
+from marshmallow import validate, Schema, fields
+from .period_schema import PeriodSchema
+
+
+class ApplicationBodySchema(Schema):
```
Comment (TeddyAndersson) [inline app/schemas/application_schema.py:5]: move to separate file, one class per file.

File: Pipfile [hunk 5]
```diff
@@ -15,6 +15,8 @@ xmltodict = "*"
 flask-restful = "*"
 hashids = "*"
 python-dotenv = "*"
+marshmallow = "*"
+marshmallow-jsonapi = "*"
```
Comment (TeddyAndersson) [inline Pipfile:19]: Remove dependency

File: app/routes/applications.py [hunk 6]
```diff
@@ -1,28 +1,89 @@
...
 
         response = application.create()
 
-        return jsonify(response)
+        response_schema = ResponseSchema()
+        try:
+            validated_response = response_schema.load(response)
+            return validated_response
+        except ValidationError as error:
+            return jsonify(error.messages)
```
Comment (TeddyAndersson) [inline app/routes/applications.py:89]: We need better error return than the default from viva.

File: app/routes/applications.py [hunk 7]
```diff
@@ -1,28 +1,89 @@
...
+            period=validated_data['period'],
         )
 
         response = application.create()
 
-        return jsonify(response)
+        response_schema = ResponseSchema()
+        try:
+            validated_response = response_schema.load(response)
+            return validated_response
```
Comment (TeddyAndersson) [inline app/routes/applications.py:87]: Better response for successful request towards viva.

**Review by TeddyAndersson — CHANGES_REQUESTED** [review]

File: Pipfile (hunk 5 above)
Comment (TeddyAndersson) [reply Pipfile:19]: marshmallow_json api is unused

File: app/schemas/period_schema.py (hunk 1 above)
Comment (dannil76) [reply app/schemas/period_schema.py]: > import fields from marshmallow and remove dependecy for marshmallow_jsonapi

Done!

File: app/schemas/response_schema.py (hunk 2 above)
Comment (dannil76) [reply app/schemas/response_schema.py]: > import fields from marshmallow and remove dependecy for marshmallow_jsonapi

Done

File: app/routes/applications.py (hunk 3 above)
Comment (dannil76) [reply app/routes/applications.py:12]: > Move to libs folder

Done

File: Pipfile (hunk 5 above)
Comment (dannil76) [reply Pipfile]: > marshmallow_json api is unused

Done

File: app/schemas/application_schema.py (hunk 4 above)
Comment (dannil76) [reply app/schemas/application_schema.py]: > move to separate file, one class per file.

done!

File: app/routes/applications.py (hunk 6 above)
Comment (dannil76) [reply app/routes/applications.py:55]: Will be done in later sprint. This will do for now. :)

File: app/routes/applications.py (hunk 7 above)
Comment (dannil76) [reply app/routes/applications.py:53]: Will be done in later sprint. This will do for now. :)

**Review by TeddyAndersson — APPROVED** [review]

**Pull request merged**